server.js

```diff
@@ -32,6 +49,38 @@ app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+app.get('/directors', async (req, res) => {
+  const directors = await Director.find()
+  res.json(directors)
+})
+
+app.get('/directors/:id', async (req, res) => {
+  const director = await Director.findById(req.params.id)
+  if (director) {
+    res.json(director)
+  } else {
+    res.status(404).json({ error: 'Author not found' })
+  }
+
+})
```

Comment:
Ok, and here's where the magic happens. Since a "director" is sort of a characteristic of a "show", a suitable RESTful endpoint could be: 
`localhost:8080/shows/directors/<the director's name>`

Seeing this, the endpoint to list directors could be something like: 

```
// ENDPOINT 2: List all shows of a certain director.
// **Example: http://localhost:8080/shows/directors/Abhishek%20Sharma**
app.get('/shows/directors/:director', async (req, res) => {
  const {director} = req.params
  const filteredShows = await Show.find({director: director})
  res.json(filteredShows)
})
```